src/net.cpp

```diff
@@ -113,6 +113,22 @@ std::map<CNetAddr, LocalServiceInfo> mapLocalHost GUARDED_BY(cs_mapLocalHost);
 static bool vfLimited[NET_MAX] GUARDED_BY(cs_mapLocalHost) = {};
 std::string strSubVersion;
 
+/**
+ * Check if making an automatic outbound connection to the given address is allowed.
+ * Could be restricted by the `-onlynet` configuration option.
+ * @return true if allowed
+ */
+static bool OutboundConnectionAllowedTo(const CNetAddr& addr)
+{
+    if (!gArgs.IsArgSet("-onlynet")) {
+        return true;
+    }
+    const auto onlynets = gArgs.GetArgs("-onlynet");
+    return std::any_of(onlynets.begin(), onlynets.end(), [&addr](const auto& net) {
```

Comment:
naming nit, `onlynet` or maybe `onlynet_str` might be a clearer iterator name for `onlynets`, and `net` is often used for passing a `Network` enum, e.g. in `src/net.h::IsReachable()` and `SetReachable()`)
```suggestion
    return std::any_of(onlynets.begin(), onlynets.end(), [&addr](const auto& onlynet) {
```

Reply:
I will optimize this because it will be called frequently (not as in https://github.com/bitcoin/bitcoin/pull/22651, once per startup). `std::unordered_set` has a lookup time `O(1)`.

Reply:
Done (optimized).